### PR TITLE
feat: add /contracts endpoint to list contracts

### PR DESCRIPTION
**Pull request by sborrazas**

### CODE REVIEW

**Review by jyeshe — COMMENTED** [review]
Looking good. I miss only `ga_attach`, `paying_for` and `ga_meta` contracts.

File: lib/ae_mdw/contracts.ex [hunk 1]
```diff
@@ -549,4 +592,18 @@ defmodule AeMdw.Contracts do
 
   defp deserialize_cursor_string(event_hash_bin),
     do: Base.decode32(event_hash_bin, padding: false)
+
+  defp serialize_contracts_cursor(nil), do: nil
+
+  defp serialize_contracts_cursor({{txi, local_idx}, is_reversed?}),
+    do: {"#{txi}-#{local_idx + 1}", is_reversed?}
```
[the remark this answers is not in the document]
Comment (sborrazas) [reply lib/ae_mdw/contracts.ex:613]: Done, added comment

**Comment by sborrazas** [comment]
> Looking good. I miss only `ga_attach`, `paying_for` and `ga_meta` contracts.

I've added the `ga_attach` transactions which map 1-to-1 to contract creations. For the inner `paying_for` transactions I wasn't able to find an existing index that would filter by `(tx_type, txi)`, so I would need to add a new one with migrations included. I'll do that on the next PR, since it involves a major individual change. Thank you for bringing it up

**Review by jyeshe — APPROVED** [review]
:pray:

**Pull request merged**